src/tests/Common/CLRTest.Execute.Bash.targets

```diff
@@ -365,6 +365,31 @@ fi
 $__Command $HARNESS_RUNNER android test --instrumentation="net.dot.MonoRunner" --package-name="net.dot.$__Category" --app="$__TestBinaryBase/$__Category.apk" --output-directory=`pwd` --arg=entrypoint:libname=$(MsBuildProjectName).dll --expected-exit-code=100 -v
 CLRTestExitCode=$?
 
+# Exist code of xharness is zero when tests finished successfully
+CLRTestExpectedExitCode=0
+      ]]>
+      </BashCLRTestLaunchCmds>
+      <BashCLRTestLaunchCmds Condition="'$(CLRTestKind)' == 'BuildAndRun' And $(TargetOS) == 'iOSSimulator' ">
+      <![CDATA[
+__Command=""
+if [ ! -z ${__TestDotNetCmd+x} ] %3B then
+  __Command+=" $__TestDotNetCmd"
+else
+  __Command+=" dotnet"
+fi
+
+if [ ! -z "$XHARNESS_CLI_PATH" ]; then
+	# When running in CI, we only have the .NET runtime available
+	# We need to call the XHarness CLI DLL directly via dotnet exec
+	HARNESS_RUNNER="exec $XHARNESS_CLI_PATH"
+else
+	HARNESS_RUNNER="xharness"
+fi
+
+mkdir -p /tmp/out
+$__Command $HARNESS_RUNNER apple run --app="$__TestBinaryBase/$__Category.app" --output-directory="/tmp/out" --set-env="MONO_APPLE_APP_ENTRY_POINT_LIB_NAME=$(MsBuildProjectName).dll" --expected-exit-code=100 -v
```

Comment:
The problem with running iOS Simulator workloads in Helix is that the way the Helix client is running doesn't work with the Simulator as there's no user GUI session. We work around this by executing the whole payload using `sudo launchctl asuser`. There are couple of more tricks like this (re-launching the `Simulator.app`, detection of Simulator hangs, cleanup..).

Since there are so many caveats and to improve stability, I have tried to capture all this in Helix SDK, specifically in these scripts:
- https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/xharness-helix-job.android.sh
- https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/xharness-helix-job.apple.sh
- https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/xharness-helix-job.apple.sh

Currently, when you define your own command and only ask for the XHarness CLI to be pre-installed, we don't then run your script as part of this wrapper and this will have to be fixed (we have an issue for that that you've seen already https://github.com/dotnet/xharness/issues/590)

Reply:
With what you try here - you will get timeouts on the simulators (the process spawns but we can't talk to it)

Reply:
I just got this work prioritized because I see it will be a blocker for you. I estimate around 4 days of work so it would be ready next week. We will then have to come back here and check how you turn your MSBuild items into `HelixWorkItem` so that we can plug it in with what I prepare.